src/randoop/util/ListOfLists.java

```diff
@@ -95,4 +98,44 @@ public T get(int index) {
   public String toString() {
     return toJDKList().toString();
   }
+
+  @Override
+  public Iterator<T> iterator() {
+    return new LOLIterator(lists.iterator());
+  }
+  
+  private class LOLIterator implements Iterator<T> {
+
+    private Iterator<SimpleList<T>> listIterator;
+    private Iterator<T> elemIterator;
+
+    public LOLIterator(Iterator<SimpleList<T>> listIterator) {
+      this.listIterator = listIterator;
+      if (this.listIterator.hasNext()) {
+        this.elemIterator = (this.listIterator).next().iterator();
+      }
+    }
+
+    @Override
+    public boolean hasNext() {
+      return (listIterator.hasNext() || (elemIterator != null && elemIterator.hasNext()));
```

Comment:
This implementation is not correct, because listIterator might contain (only) an empty iterator, in which case the LOLIterator doesn't acutally contain any elements and hasNext should return false.